components/tooltip/tooltip.js

```diff
@@ -725,15 +725,23 @@ class Tooltip extends RtlMixin(LitElement) {
 	_findTarget() {
 		const ownerRoot = this.getRootNode();
 
-		let target;
+		let target, targetSelector;
 		if (this.for) {
-			const targetSelector = `#${cssEscape(this.for)}`;
+			targetSelector = `#${cssEscape(this.for)}`;
 			target = ownerRoot.querySelector(targetSelector);
-			target = target || (ownerRoot && ownerRoot.host && ownerRoot.host.querySelector(targetSelector));
+			target = target || ownerRoot?.host?.querySelector(targetSelector);
 		} else {
+			console.warn('<d2l-tooltip>: missing required attribute "for"');
 			const parentNode = this.parentNode;
 			target = parentNode.nodeType === Node.DOCUMENT_FRAGMENT_NODE ? ownerRoot.host : parentNode;
 		}
+
+		if (window.getComputedStyle(target).position === 'static') {
+			const host = this.getRootNode().host?.tagName.toLowerCase() ?? 'body';
+			const selectorText = targetSelector ? `(${targetSelector}) ` : '';
+			console.warn(`In <${host}>, <d2l-tooltip> target element ${selectorText}does not have a CSS position. The tooltip may not position itself correctly on the page.`);
```

Comment:
I added the `host` here mostly for visibility while fixing tests. I don't think it's necessary to always include, though.